src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -550,21 +550,15 @@ protected override void ProcessRecord()
                         FillRequestStream(request);
                         try
                         {
-                            long requestContentLength = request.Content is null ? 0 : request.Content.Headers.ContentLength.Value;
-
-                            string reqVerboseMsg = string.Format(
-                                CultureInfo.CurrentCulture,
-                                WebCmdletStrings.WebMethodInvocationVerboseMsg,
-                                request.Version,
-                                request.Method,
-                                requestContentLength);
-
-                            WriteVerbose(reqVerboseMsg);
-
                             _maximumRedirection = WebSession.MaximumRedirection;
 
                             using HttpResponseMessage response = GetResponse(client, request, handleRedirect);
 
+                            long requestContentLength = request.Content is null ? 0 : request.Content.Headers.ContentLength.Value;
```

Comment:
But still _request_ length?

Reply:
Yes, the response length is in the next verbose message:
```cs
                            string contentType = ContentHelper.GetContentType(response);
                            long? contentLength = response.Content.Headers.ContentLength;
                            string respVerboseMsg = contentLength is null
                                ? string.Format(CultureInfo.CurrentCulture, WebCmdletStrings.WebResponseNoSizeVerboseMsg, contentType)
                                : string.Format(CultureInfo.CurrentCulture, WebCmdletStrings.WebResponseVerboseMsg, contentLength, contentType);
                            
                            WriteVerbose(respVerboseMsg)
```

Reply:
Why then mix what was before and after?

Reply:
We can't have the correct info on the HTTP version from the `request`, only from the `response`. I think the verbose message looks better as it is instead of moving the HTTP version to the `respVerboseMsg`. Maybe we could redesign the verbose messages in a better way, I welcome your suggestions @mklement0 and @iSazonov

Reply:
It would be useful to see the initial parameters before the request, especially the implicit (default) values and the actual values after the request.

Reply:
I agree that we should have two messages: request and response.  Mixing the two makes the verbose message confusing to someone trying to rely on it for troubleshooting.